container/lxd/initialisation_linux.go

```diff
@@ -304,47 +308,38 @@ func ensureDependencies(series string) error {
 //
 // TODO(frobware): this only caters for IPv4 setups.
 func findNextAvailableIPv4Subnet() (string, error) {
-	_, ip10network, err := net.ParseCIDR("10.0.0.0/16")
-	if err != nil {
-		return "", errors.Trace(err)
-	}
-
 	addrs, err := interfaceAddrs()
 	if err != nil {
 		return "", errors.Annotatef(err, "cannot get network interface addresses")
 	}
 
-	max := 0
-	usedSubnets := make(map[int]bool)
-
-	for _, address := range addrs {
-		addr, network, err := net.ParseCIDR(address.String())
+	randomized3rdSegment := randomizedOctetRange()
+	for _, i := range randomized3rdSegment {
+		// lxd randomizes the 2nd and 3rd segments, we should be fine with the
+		// 3rd only
+		_, ip10network, err := net.ParseCIDR(fmt.Sprintf("10.0.%d.0/24", i))
 		if err != nil {
-			logger.Debugf("cannot parse address %q: %v (ignoring)", address.String(), err)
-			continue
-		}
-		if !ip10network.Contains(addr) {
-			logger.Debugf("find available subnet, skipping %q", network.String())
-			continue
-		}
-		subnet := int(network.IP[2])
-		usedSubnets[subnet] = true
-		if subnet > max {
-			max = subnet
+			return "", errors.Trace(err)
 		}
-	}
 
-	if len(usedSubnets) == 0 {
-		return "0", nil
-	}
+		collides := false
+		for _, address := range addrs {
+			addr, network, err := net.ParseCIDR(address.String())
+			if err != nil {
+				logger.Debugf("cannot parse address %q: %v (ignoring)", address.String(), err)
+				continue
+			}
 
-	for i := 0; i < 256; i++ {
-		max = (max + 1) % 256
-		if _, inUse := usedSubnets[max]; !inUse {
-			return fmt.Sprintf("%d", max), nil
+			if ip10network.Contains(addr) {
```

Comment:
We need to do this the other way around as well. If the existing address's network contains the proposed subnet IP. (eg we have an address 10.0.1.3 but the subnet is 10.0.0.0/22 We shouldn't use 10.0.2.0/24 as they overlap.)